### DISCUSSION TITLE
empty json request when accessing information

### DISCUSSION THREAD

**Discussion by VinylVault** [joalla/discogs_client#26] [Ideas]
can we please ensure that and API request recieves valid JSON back before allowing it to be exposed to users code

**Comment by alifhughes** [comment]
Hi @VinylVault, Please can you give some examples of when this has happened to you and under what circumstances? It will make it a lot easier to reproduce the problem if we have these. From there we can see if it is an issue that needs addressing. Thanks

**Reply by VinylVault** [reply]
basically it was just a collection import, to a local database, ive seen it on image requests, and release data requests, problem is its random as to when it happens, sometimes it will happen 100 releases into the import, sometimes 1000 there is no particular pattern sadly which means tracking it down will be a nightmare, which is why i think, and suggested just passing the json through a confirm valid json gate befre releasing it to the rest of the code

**Reply by alifhughes** [reply]
Okay I'll try to look into it but feel free to make a PR with the suggestion, this is an open source project after all and we always welcome contributions.

**Reply by VinylVault** [reply]
It's just happened again, here's the console output of the error, BTW it has successfully completed the record it failed on in the past

`
  File "discogsimport.py", line 231, in <module>
    if not company.images:
  File "/home/dex/.local/lib/python3.7/site-packages/discogs_client/models.py", line 26, in __get__
    value = instance.fetch(self.name)
  File "/home/dex/.local/lib/python3.7/site-packages/discogs_client/models.py", line 237, in fetch
    self.refresh()
  File "/home/dex/.local/lib/python3.7/site-packages/discogs_client/models.py", line 203, in refresh
    data = self.client._get(self.data['resource_url'])
  File "/home/dex/.local/lib/python3.7/site-packages/discogs_client/client.py", line 112, in _get
    return self._request('GET', url)
  File "/home/dex/.local/lib/python3.7/site-packages/discogs_client/client.py", line 104, in _request
    body = json.loads(content)
  File "/usr/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.7/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
`

**Reply by VinylVault** [reply]
Once i get my head around the code, I'll try and code a PR to check the JSON and re-request it if its invalid, might take some time as I'm new to python

**Reply by VinylVault** [reply]
After a few more of these errors, I think I've tracked it down to not being a discogs_client specfic error, more its when the connection to Discogs falls over, we all know how unstable Discogs has been over the last little while - i guess stores are moving a lot of inventory onto the site seeing as they cant have shoppers in the stores
